petastorm/spark/spark_dataset_converter.py

```diff
@@ -517,6 +535,19 @@ def wait_for_file(path):
         pool.join()
 
 
+def _check_dataset_file_median_size(url_list):
+    fs, path_list = get_filesystem_and_path_or_paths(url_list)
+    # TODO: also check other file system.
+    if isinstance(fs, LocalFileSystem):
+        file_size_list = [os.path.getsize(path) for path in path_list]
+        mid_index = len(file_size_list) // 2
+        median_size = sorted(file_size_list)[mid_index]
+        if median_size < 50 * 1024 * 1024:
```

Comment:
number of files being small should not be an issue. Because when reading, each file will be split into row-groups, one row-group is a minimal reading task.